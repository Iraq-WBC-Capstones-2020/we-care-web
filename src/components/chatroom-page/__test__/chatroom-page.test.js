// const add = (x, y) => x + y;

// const add = jest.fn(() => 3);

// test('add', () => {
//   const value = add(1, 2);
//   expect(add).toHaveBeenCalledTimes(1);
//   expect(value).toBe(3);
// });

// const total = (subTotal, total) => {
//   return '$' + add(subTotal, total);
// };

// test('total', () => {
//   expect(total(5, 20)).toBe('$25');
// });

import React from 'react';
import ChatroomPage from './../chatroom-page';
import renderer from 'react-test-renderer';
import { BrowserRouter as Router } from 'react-router-dom';

test('renders correctly', () => {
  const component = renderer
    .create(
      <Router>
        <ChatroomPage />
      </Router>
    )
    .toJSON();
  expect(component).toMatchSnapshot();
});
