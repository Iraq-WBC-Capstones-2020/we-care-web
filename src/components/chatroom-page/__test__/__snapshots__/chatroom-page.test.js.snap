// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div
  className="h-screen flex flex-col bg-darkP"
>
  <nav
    className="relative flex flex-wrap items-center justify-between px-2 lg:py-6 py-4 navbar-expand-lg bg-darkP text-beige text-sm"
  >
    <div
      className="container px-4 mx-auto flex flex-wrap items-center justify-between"
    >
      <div
        className="w-full relative flex justify-between lg:w-auto lg:static lg:block lg:justify-start"
      >
        <img
          alt="We Care logo"
          className="w-16 md:w-20 lg:ml-0 md:ml-12 ml-8 cursor-pointer"
          src="Logo.svg"
        />
        <button
          className="text-white cursor-pointer text-xl leading-none px-3 py-1 border border-solid border-transparent rounded bg-transparent block lg:hidden outline-none focus:outline-none"
          onClick={[Function]}
          type="button"
        >
          <svg
            className="text-2xl text-beige"
            fill="none"
            height="1em"
            stroke="currentColor"
            strokeLinecap="round"
            strokeLinejoin="round"
            strokeWidth="2"
            style={
              Object {
                "color": undefined,
              }
            }
            viewBox="0 0 24 24"
            width="1em"
            xmlns="http://www.w3.org/2000/svg"
          >
            <line
              x1="3"
              x2="21"
              y1="12"
              y2="12"
            />
            <line
              x1="3"
              x2="21"
              y1="6"
              y2="6"
            />
            <line
              x1="3"
              x2="21"
              y1="18"
              y2="18"
            />
          </svg>
        </button>
      </div>
      <div
        className="lg:flex flex-grow items-center justify-center hidden"
        id="example-navbar-danger"
      >
        <ul
          className="flex flex-col justify-center lg:flex-row list-none text-center w-full"
        >
          <li
            className="lg:mr-10 lg:mt-0 lg:py-0 py-3 hover:text-orangeP lg:border-0 border-b border-beige"
          >
            <a
              href="/profile/feed"
              onClick={[Function]}
            >
              Profile
            </a>
          </li>
          <li
            className="lg:mr-10 lg:my-0 lg:py-0 py-3 hover:text-orangeP lg:border-0 border-b border-beige"
          >
            <a
              href="#"
            >
              For you
            </a>
          </li>
          <li
            className="lg:mb-0 lg:py-0 py-3 hover:text-orangeP"
          >
            <a
              href="#"
            >
              Counseling
            </a>
          </li>
        </ul>
      </div>
      <div
        className="lg:flex justify-center items-center hidden"
      >
        <a
          href="/profile"
          onClick={[Function]}
        >
          <img
            className="rounded-full h-10 w-10 object-cover"
            src="profile.png"
          />
        </a>
        <a
          href="/profile"
          onClick={[Function]}
        >
          <p
            className="ml-2"
          >
            Bruce Lee
          </p>
        </a>
      </div>
    </div>
  </nav>
  <main
    className="flex lg:justify-evenly justify-center items-center h-full w-screen overflow-hidden"
  >
    <img
      className="w-1/4 hidden lg:inline-block"
      src="chat-undraw.svg"
    />
    <div
      className="flex flex-col justify-between md:w-10/12 w-11/12 lg:w-7/12 bg-beige m-5 rounded-lg relative overflow-y-auto overflow-x-hidden"
      style={
        Object {
          "height": "90%",
        }
      }
    >
      <button
        className="focus:outline-none sticky top-0 "
      >
        <svg
          className="text-darkP text-2xl"
          fill="currentColor"
          height="1em"
          stroke="currentColor"
          strokeWidth="0"
          style={
            Object {
              "color": undefined,
            }
          }
          viewBox="0 0 512 512"
          width="1em"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M256 48C141.1 48 48 141.1 48 256s93.1 208 208 208 208-93.1 208-208S370.9 48 256 48zm52.7 283.3L256 278.6l-52.7 52.7c-6.2 6.2-16.4 6.2-22.6 0-3.1-3.1-4.7-7.2-4.7-11.3 0-4.1 1.6-8.2 4.7-11.3l52.7-52.7-52.7-52.7c-3.1-3.1-4.7-7.2-4.7-11.3 0-4.1 1.6-8.2 4.7-11.3 6.2-6.2 16.4-6.2 22.6 0l52.7 52.7 52.7-52.7c6.2-6.2 16.4-6.2 22.6 0 6.2 6.2 6.2 16.4 0 22.6L278.6 256l52.7 52.7c6.2 6.2 6.2 16.4 0 22.6-6.2 6.3-16.4 6.3-22.6 0z"
          />
        </svg>
      </button>
      <div
        className="w-11/12 self-center flex flex-col"
      >
        <div
          className="bg-white w-10/12 md:w-7/12 lg:w-7/12 xl:w-5/12 rounded-md text-darkP flex flex-col md:p-5 p-3 my-4 self-start"
        >
          <div
            className="flex mb-3"
          >
            <div>
              <h2
                className="md:text-base text-sm font-semibold"
              >
                wonderful-sky
              </h2>
              <p
                className="text-xs md:text-sm text-orangeP"
              >
                2:45 PM
              </p>
            </div>
          </div>
          <div
            className="mb-3 text-xs md:text-sm"
          >
            Lorem ipsum dolor sit amet, consectetur adipiscing elit.
          </div>
        </div>
        <div
          className="bg-white w-10/12 md:w-7/12 lg:w-7/12 xl:w-5/12 rounded-md text-darkP flex flex-col md:p-5 p-3 my-4 self-end"
        >
          <div
            className="flex mb-3"
          >
            <div>
              <h2
                className="md:text-base text-sm font-semibold"
              >
                Kind-heart
              </h2>
              <p
                className="text-xs md:text-sm text-orangeP"
              >
                2:47 PM
              </p>
            </div>
          </div>
          <div
            className="mb-3 text-xs md:text-sm"
          >
            Lorem ipsum dolor sit amet, consectetur adipiscing elit. Lorem ipsum dolor sit amet.
          </div>
        </div>
        <div
          className="bg-white w-10/12 md:w-7/12 lg:w-7/12 xl:w-5/12 rounded-md text-darkP flex flex-col md:p-5 p-3 my-4 self-start"
        >
          <div
            className="flex mb-3"
          >
            <div>
              <h2
                className="md:text-base text-sm font-semibold"
              >
                wonderful-sky
              </h2>
              <p
                className="text-xs md:text-sm text-orangeP"
              >
                2:50 PM
              </p>
            </div>
          </div>
          <div
            className="mb-3 text-xs md:text-sm"
          >
            Lorem ipsum dolor sit amet, consectetur adipiscing elit. Lorem ipsum dolor sit amet.
          </div>
        </div>
      </div>
      <div
        className="w-11/12 text-darkP relative my-10 self-center"
      >
        <textarea
          className="resize-none rounded-md text-darkP px-4 py-3 h-32 w-full mt-4 md:text-base text-sm"
          id="text-area"
          placeholder="Start typing..."
        />
        <button
          className="bg-darkP flex items-center text-beige rounded-sm md:px-6 px-3 py-1 absolute bottom-0 right-0 mr-4 mb-4 hover:text-orangeP"
          onClick={[Function]}
        >
          <p
            className="md:text-base text-sm"
          >
            Send
          </p>
          <svg
            className="ml-2"
            fill="none"
            height="1em"
            stroke="currentColor"
            strokeLinecap="round"
            strokeLinejoin="round"
            strokeWidth="2"
            style={
              Object {
                "color": undefined,
              }
            }
            viewBox="0 0 24 24"
            width="1em"
            xmlns="http://www.w3.org/2000/svg"
          >
            <line
              x1="22"
              x2="11"
              y1="2"
              y2="13"
            />
            <polygon
              points="22 2 15 22 11 13 2 9 22 2"
            />
          </svg>
        </button>
      </div>
    </div>
  </main>
</div>
`;
